### PR TITLE
Upgrade hadoop version from 2.7.3 to 3.3.0

### PR DESCRIPTION
**Pull request by bradyoo**
All dependencies to it appear to be internal

Didn't touch underfs/hdfs/pom.xml's hadoop version (changing this to 3.3.0 causes breakage)

### CODE REVIEW

**Comment by yuzhu** [comment]
@calvinjia @apc999 PTAL at the hdfs client dependency change. Does this require any changes in generateTarball script?

**Comment by apc999** [comment]
yes we need to update the generate-tarball script
please also check underfs-hdfs module and I believe there are some macros checking Hadoop versions and switch code there.

taking a step back, @bradyoo  @calvinjia  what's the motivation of bumping hadoop version to 3.3 ?

**Comment by yuzhu** [comment]
java 11 support

File: shaded/hadoop/pom.xml [hunk 1]
```diff
@@ -27,7 +27,7 @@
     <!-- The build path need to be defined here as well as in the parent pom so that mvn can -->
     <!-- run properly from sub-project directories -->
     <build.path>${project.parent.parent.basedir}/build</build.path>
-    <ufs.hadoop.version>2.7.3</ufs.hadoop.version>
+    <ufs.hadoop.version>3.3.0</ufs.hadoop.version>
```
Comment (calvinjia) [inline shaded/hadoop/pom.xml:30]: Do we need to update the default profile to ufs-hadoop-3?

File: pom.xml [hunk 2]
```diff
@@ -132,7 +132,7 @@
     <grpc.version>1.28.1</grpc.version>
     <netty.version>4.1.45.Final</netty.version>
     <rocksdb.version>5.15.10</rocksdb.version>
-    <hadoop.version>2.7.3</hadoop.version>
+    <hadoop.version>3.3.0</hadoop.version>
```
Comment (calvinjia) [inline pom.xml:135]: Does this cause any issues with the Hadoop compatible API? Is the FileSystem interface backward compatible?

**Comment by yuzhu** [comment]
included in #11851 . Closing this for now

**Pull request closed without merging**